app/grandchallenge/algorithms/tasks.py

```diff
@@ -21,6 +23,100 @@
 from grandchallenge.subdomains.utils import reverse
 
 
+@shared_task
+def create_algorithm_job_for_inputs(
+    *, algorithm_image_pk, civ_pks, upload_pks, creator_pk
+):
+    try:
+        algorithm_image = AlgorithmImage.objects.get(pk=algorithm_image_pk)
+    except AlgorithmImage.DoesNotExist:
+        return
+
+    if Job.objects.filter(
+        algorithm_image=algorithm_image,
+        inputs__pk__in=civ_pks,
+        creator_id=creator_pk,
+    ).exists():
+        return
+
+    # Editors group should be able to view session jobs for debugging
+    groups = [algorithm_image.algorithm.editors_group]
+
+    job = create_algorithm_job_with_inputs(
+        algorithm_image=algorithm_image,
+        inputs=civ_pks,
+        creator=get_user_model().objects.get(pk=creator_pk),
+        extra_viewer_groups=groups,
+    )
+    start_jobs = execute_algorithm_job_for_inputs.signature(
+        kwargs={"job_pk": job.pk}, immutable=True
+    )
+    if upload_pks:
+        image_tasks = group(
+            create_component_interface_value_for_image.signature(
+                kwargs={
+                    "component_interface_value_pk": civ_pk,
+                    "upload_pk": upload_pk,
+                },
+                immutable=True,
+            )
+            for civ_pk, upload_pk in upload_pks.items()
+        )
+        start_jobs = chord(image_tasks, start_jobs)
+
+    start_jobs.apply_async()
+    return job
+
+
+@shared_task
+def create_component_interface_value_for_image(
```

Comment:
The name of this function is confusing as the component interface value has already been created, this is building the images and then attaching it to the CIV.